hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/formats/v6/SignatureWriterV6.java

```diff
@@ -0,0 +1,150 @@
+            @NonNull final Bytes endRunningHash) {
+        // write signature file
+        final var sigFilePath = getSigFilePath(recordFilePath);
+        try (final var fileOut = Files.newOutputStream(sigFilePath, StandardOpenOption.CREATE_NEW)) {
+            final var streamingData = new WritableStreamingData(fileOut);
+            Bytes metadataHash = null;
+            if (writeMetadataSignature) {
+                // create metadata hash
+                HashingOutputStream hashingOutputStream =
+                        new HashingOutputStream(MessageDigest.getInstance(DigestType.SHA_384.algorithmName()));
+                SerializableDataOutputStream dataOutputStream = new SerializableDataOutputStream(hashingOutputStream);
+                dataOutputStream.writeInt(recordFileVersion);
+                dataOutputStream.writeInt(hapiProtoVersion.major());
+                dataOutputStream.writeInt(hapiProtoVersion.minor());
+                dataOutputStream.writeInt(hapiProtoVersion.patch());
+                startRunningHash.writeTo(dataOutputStream);
+                endRunningHash.writeTo(dataOutputStream);
+                dataOutputStream.writeLong(blockNumber);
+                dataOutputStream.close();
+                metadataHash = Bytes.wrap(hashingOutputStream.getDigest());
+            }
+            // create signature file
+            final var signatureFile = new SignatureFile(
+                    generateSignatureObject(signer, recordFileHash),
+                    writeMetadataSignature ? generateSignatureObject(signer, metadataHash) : null);
+            // write version in signature file. It is only 1 byte, compared to 4 in record files
+            streamingData.writeByte((byte) 6);
+            // write protobuf SignatureFile
+            SignatureFile.PROTOBUF.write(signatureFile, streamingData);
+            logger.debug("signature file saved: {}", sigFilePath);
+            // flush
+            fileOut.flush();
+        } catch (final IOException e) {
+            logger.error("Fail to generate signature file for {}", recordFilePath, e);
+            throw new UncheckedIOException(e);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Given a hash produce the HAPI signature object for that hash
+     *
+     * @param signer The signer to use
+     * @param hash The hash to sign
+     * @return Signature object
+     */
+    private static SignatureObject generateSignatureObject(@NonNull final Signer signer, @NonNull final Bytes hash) {
+        final Bytes signature = Bytes.wrap(signer.sign(hash.toByteArray()).getSignatureBytes());
+        return SignatureObject.newBuilder()
+                .type(SignatureType.SHA_384_WITH_RSA)
+                .length((int) signature.length())
+                .checksum(101 - (int) signature.length()) // simple checksum to detect if at wrong place in
```

Comment:
I guess we'll never know where this `101` came from ...

Reply:
True 🤣